js/src/modals/ExportAccount/exportStore.js

```diff
@@ -0,0 +1,107 @@
+// Copyright 2015-2017 Parity Technologies (UK) Ltd.
+// This file is part of Parity.
+
+// Parity is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity.  If not, see <http://www.gnu.org/licenses/>.
+
+import { action, observable } from 'mobx';
+import FileSaver from 'file-saver';
+
+export default class ExportStore {
+  @observable canExport = false;
+  @observable selectedAccount = '';
+  @observable selectedAccounts = {};
+  @observable accountValue = '';
+  @observable inputValue = {};
+
+  constructor (api, accounts, newError, address) {
+    this._api = api;
+    this._accounts = accounts;
+    this._newError = newError;
+    this._address = address;
+  }
+
+  @action changePassword = (event, password) => {
+    const selectedAccount = (this._address) ? null : this.selectedAccount;
+
+    this.setPassword(selectedAccount, password);
+  }
+
+  @action getPassword = (account) => {
+    return this.inputValue[account];
+  }
+
+  @action onFocus = (address) => {
+    this.selectedAccount = address;
+  }
+
+  @action resetAccountValue = () => {
+    this.accountValue = '';
+  }
+
+  @action setPassword = (account, password) => {
+    (this._address)
+      ? this.accountValue = password
+      : this.inputValue[account] = password;
+  }
+
+  @action setSelectedAccount = (addr) => {
+    this.selectedAccounts[addr] = true;
+    this.canExport = true;
+  }
+
+  @action toggleSelectedAccount = (addr) => {
+    this.selectedAccounts[addr] = !this.selectedAccounts[addr];
+    if (!this.selectedAccounts[addr]) {
+      delete this.selectedAccounts[addr];
+    }
+    this.canExport = false;
+    Object
+      .keys(this.selectedAccounts)
+      .forEach((address) => {
+        if (this.selectedAccounts[address]) {
+          this.canExport = true;
+        }
+      });
```

Comment:
This is the spookiest bug I've come across. If you know the answer you're a god send.

If I don't iterate over the keys, mobx won't trigger an state update for react. I did it your way the first time... I don't remember where I read about this bug and I don't know what's happening under the hood. If you have advice about how to avoid it in the future I would be super appreciative.